Appium test reports

clample opened the pull request.

Commit:
Add initial documentation for Appium Test Reports

Commit:
Make test reports pages framework specefic

Add documentation specifically for Appium test reports.

ragog commented on docs/tools/appium/test-reports.md:
```diff
@@ -0,0 +1,55 @@
+
+<h3 id="sharing-test-reports">Sharing Test Reports</h3>
+
+In order to share the test report with others, simply select the Share button from your test report. This will generate a link which you can share with other team members, giving them access to the full Test Report. This feature can be useful when trying to show a developer a failing test, or simply to showcase new changes to the application.
```
Users can share the report not only with other team members, but with whoever has the link (also unregistered users).

Commit:
Remove dashboard links

The dashboard page will be removed in the help page refractoring. We
shouldn't link to it.

Commit:
Update sharing feature

The user doesn't need to be a team member to view the link. Also, no
account is required.

Commit:
Improve test report introduction

Explain why having test data organised is valuable.

Commit:
Merge branch 'master' into appium-test-reports

Commit:
Fix broken links
